### PR TITLE
Clean up transport versions 8,500,000-8,500,009

### PR DESCRIPTION
**Pull request by thecoop**
Collapse transport versions 8.9.0 and 8,500,000-8,500,009 to 8,500,010

### CODE REVIEW

**Comment by thecoop** [comment]
@elasticmachine rerun 
elasticsearch-ci/packaging-tests-windows-sample

File: server/src/main/java/org/elasticsearch/TransportVersion.java [hunk 1]
```diff
@@ -123,22 +123,10 @@ private static TransportVersion registerTransportVersion(int id, String uniqueId
     public static final TransportVersion V_8_8_0 = registerTransportVersion(8_08_00_99, "f64fe576-0767-4ec3-984e-3e30b33b6c46");
     public static final TransportVersion V_8_8_1 = registerTransportVersion(8_08_01_99, "291c71bb-5b0a-4b7e-a407-6e53bc128d0f");
 
-    public static final TransportVersion V_8_9_0 = registerTransportVersion(8_09_00_99, "13c1c2cb-d975-461f-ab98-309ebc1c01bc");
```
Comment (ChrisHegarty) [inline server/src/main/java/org/elasticsearch/TransportVersion.java:126]: How  can we remove this released ES version? Even if not used, should we not keep it?

File: server/src/main/java/org/elasticsearch/TransportVersion.java [hunk 2]
```diff
@@ -123,22 +123,10 @@ private static TransportVersion registerTransportVersion(int id, String uniqueId
...
-    public static final TransportVersion V_8_500_002 = registerTransportVersion(8_500_002, "055dd314-ff40-4313-b4c6-9fccddfa42a8");
-    public static final TransportVersion V_8_500_003 = registerTransportVersion(8_500_003, "30adbe0c-8614-40dd-81b5-44e9c657bb77");
-    public static final TransportVersion V_8_500_004 = registerTransportVersion(8_500_004, "6a00db6a-fd66-42a9-97ea-f6cc53169110");
-    public static final TransportVersion V_8_500_005 = registerTransportVersion(8_500_005, "65370d2a-d936-4383-a2e0-8403f708129b");
-    public static final TransportVersion V_8_500_006 = registerTransportVersion(8_500_006, "7BB5621A-80AC-425F-BA88-75543C442F23");
-    public static final TransportVersion V_8_500_007 = registerTransportVersion(8_500_007, "77261d43-4149-40af-89c5-7e71e0454fce");
-    // Introduced for stateless plugin
-    public static final TransportVersion V_8_500_008 = registerTransportVersion(8_500_008, "8884ab9d-94cd-4bac-aff8-01f2c394f47c");
-    public static final TransportVersion V_8_500_009 = registerTransportVersion(8_500_009, "35091358-fd41-4106-a6e2-d2a1315494c1");
```
Comment (ChrisHegarty) [inline server/src/main/java/org/elasticsearch/TransportVersion.java:141]: Is there any obvious way to determine the range that can safely be removed?

File: server/src/main/java/org/elasticsearch/TransportVersion.java (hunk 1 above)
Comment (ChrisHegarty) [reply server/src/main/java/org/elasticsearch/TransportVersion.java:126]: Ah ok, that is indeed misleading. For convenience, should we not have a constant that encodes this? Or maybe that is just not useful at this point? e.g.

```
public static final TransportVersion V_8_9_0 = V_8_500_020;
```

File: server/src/main/java/org/elasticsearch/TransportVersion.java (hunk 2 above)
Comment (ChrisHegarty) [reply server/src/main/java/org/elasticsearch/TransportVersion.java:141]: Ah ok, so this is 8.9 - 10 prior TA versions?  I guess we could be a little more aggressive here and remove all versions prior to 8.9.   But I'm ok with being a little conservative here, the all-prior-to-8.9 can be removed next week (or so)?

Comment (thecoop) [reply server/src/main/java/org/elasticsearch/TransportVersion.java:141]: The first 10 seem like an easy target for the first removal

File: server/src/main/java/org/elasticsearch/TransportVersion.java (hunk 1 above)
Comment (thecoop) [reply server/src/main/java/org/elasticsearch/TransportVersion.java:126]: There's a comment above the TV that was used whan a release was branched

**Review by ChrisHegarty — APPROVED** [review]
LTGM

**Pull request merged**